dotCMS/src/main/java/com/dotcms/webdav/FsDirectoryResource.java

```diff
@@ -0,0 +1,263 @@
+import com.dotcms.repackage.com.bradmcevoy.http.exceptions.NotAuthorizedException;
+import com.dotmarketing.util.Logger;
+
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+
+/**
+ * Represents a directory in a physical file system.
+ *
+ */
+public class FsDirectoryResource extends FsResource implements MakeCollectionableResource, PutableResource, CopyableResource, DeletableResource, MoveableResource, PropFindableResource, LockingCollectionResource, GetableResource {
+
+
+    
+    private final FileContentService contentService;
+
+    public FsDirectoryResource(String host, FileSystemResourceFactory factory, File dir, FileContentService contentService) {
+        super(host, factory, dir);
+        this.contentService = contentService;
+        if (!dir.exists()) {
+            throw new IllegalArgumentException("Directory does not exist: " + dir.getAbsolutePath());
+        }
+        if (!dir.isDirectory()) {
+            throw new IllegalArgumentException("Is not a directory: " + dir.getAbsolutePath());
+        }
+    }
+
+    @Override
+    public CollectionResource createCollection(String name) {
+        File fnew = new File(file, name);
+        boolean ok = fnew.mkdir();
+        if (!ok) {
+            throw new RuntimeException("Failed to create: " + fnew.getAbsolutePath());
+        }
+        return new FsDirectoryResource(host, factory, fnew, contentService);
+    }
+
+    @Override
+    public Resource child(String name) {
+        File fchild = new File(file, name);
+        return factory.resolveFile(this.host, fchild);
+
+    }
+
+    @Override
+    public List<? extends Resource> getChildren() {
+        ArrayList<FsResource> list = new ArrayList<FsResource>();
+        File[] files = this.file.listFiles();
+        if (files != null) {
+            for (File fchild : files) {
+                FsResource res = factory.resolveFile(this.host, fchild);
+                if (res != null) {
+                    list.add(res);
+                } else {
+                  Logger.info(FileSystemResourceFactory.class,"Couldnt resolve file " + fchild.getAbsolutePath());
+                }
+            }
+        }
+        return list;
+    }
+
+    /**
+     * Will redirect if a default page has been specified on the factory
+     *
+     * @param request
+     * @return
+     */
+    @Override
+    public String checkRedirect(Request request) {
+        if (factory.getDefaultPage() != null) {
+            return request.getAbsoluteUrl() + "/" + factory.getDefaultPage();
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public Resource createNew(String name, InputStream in, Long length, String contentType) throws IOException {
+        File dest = new File(this.getFile(), name);
+        FileOutputStream out = null;
+        try {

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Change this "try" to a try-with-resources. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2093)